src/rdb.c

```diff
@@ -3450,10 +3450,9 @@ static void backgroundSaveDoneHandlerSocket(int exitcode, int bysignal) {
     if (!bysignal && exitcode == 0) {
         serverLog(LL_NOTICE, "Background RDB transfer terminated with success");
     } else if (!bysignal && exitcode != 0) {
-        serverLog(LL_WARNING, "Background transfer error");
-        server.lastbgsave_status = C_ERR;
```

Comment:
IMHO we should maintain the current use of metrics in our tests while stabilazing the branch. Replacing metrics with logs at this stage could potentially introduce unnecessary complexity.
I do, however, agree that we need a more appropriate way to track diskless bgsave errors. Instead of repurposing the existing `lastbgsave_status`.

Reply:
The complexity i think is ok since i think the logs can do the same thing. yes, we can re-visit this after your fix get merged and see if the tests will be broken with this changes.

I think we need to set up some of the same fields separately for diskless. i am waiting for other people's opinions (and the fields names, the name seems difficult to pick)